src/box3d.rs

```diff
@@ -0,0 +1,1044 @@
+            && self.max_x() > other.min_x()
+            && self.min_y() < other.max_y()
+            && self.max_y() > other.min_y()
+            && self.min_z() < other.max_z()
+            && self.max_z() > other.min_z()
+    }
+
+    #[inline]
+    pub fn intersection(&self, other: &Self) -> Option<Self> {
+        if !self.intersects(other) {
+            return None;
+        }
+
+        let intersection_min = TypedPoint3D::new(
+            max(self.min_x(), other.min_x()),
+            max(self.min_y(), other.min_y()),
+            max(self.min_z(), other.min_z()),
+        );
+
+        let intersection_max = TypedPoint3D::new(
+            min(self.max_x(), other.max_x()),
+            min(self.max_y(), other.max_y()),
+            min(self.max_z(), other.max_z()),
+        );
+
+        Some(TypedBox3D::new(
+            intersection_min, 
+            intersection_max,
+        ))
+    }
+}
+
+impl<T, U> TypedBox3D<T, U>
+where
+    T: Copy + Add<T, Output = T>,
+{
+    /// Returns the same box3d, translated by a vector.
+    #[inline]
+    #[cfg_attr(feature = "unstable", must_use)]
+    pub fn translate(&self, by: &TypedVector3D<T, U>) -> Self {
+        Self::new(self.min + *by, self.max + *by)
+    }
+}
+
+impl<T, U> TypedBox3D<T, U>
+where
+    T: Copy + PartialOrd + Zero,
+{
+    /// Returns true if this box3d contains the point. Points are considered
+    /// in the box3d if they are on the front, left or top faces, but outside if they
+    /// are on the back, right or bottom faces.
+    #[inline]
+    pub fn contains(&self, other: &TypedPoint3D<T, U>) -> bool {
+        self.min_x() <= other.x && other.x < self.max_x()
+            && self.min_y() < other.y && other.y <= self.max_y()
+            && self.min_z() < other.z && other.z <= self.max_z()
+    }
+}
+
+impl<T, U> TypedBox3D<T, U>
+where
+    T: Copy + PartialOrd + Zero + Sub<T, Output = T>,
+{
+    /// Returns true if this box3d contains the interior of the other box3d. Always
+    /// returns true if other is empty, and always returns false if other is
+    /// nonempty but this box3d is empty.
+    #[inline]
+    pub fn contains_box(&self, other: &Self) -> bool {
+        other.is_empty()
+            || (self.min_x() <= other.min_x() && other.max_x() <= self.max_x()
+                && self.min_y() <= other.min_y() && other.max_y() <= self.max_y()
+                && self.min_z() <= other.min_z() && other.max_z() <= self.max_z())
+    }
+}
+
+impl<T, U> TypedBox3D<T, U>
+where
+    T: Copy + Sub<T, Output = T>,
+{
+    #[inline]
+    pub fn size(&self)-> TypedSize3D<T, U> {
+        TypedSize3D::new(
+            self.max_x() - self.min_x(),
+            self.max_y() - self.min_y(),
+            self.max_z() - self.min_z(),
+        )
+    }
+}
+
+impl<T, U> TypedBox3D<T, U>
+where
+    T: Copy + PartialEq + Add<T, Output = T> + Sub<T, Output = T> + Div<T, Output = T> + One,
+{
+    /// Inflates the box by the specified sizes.
```

Comment:
Do you have a better suggestion on how to word this?

Reply:
This looks reasonable.